x/tally/keeper/endblock.go

```diff
@@ -210,93 +201,39 @@ func (k Keeper) FilterAndTally(ctx sdk.Context, req types.Request) (TallyResult,
 	}
 
 	// Phase I: Filtering
+	var filterResult FilterResult
 	filter, err := k.BuildFilter(ctx, req.ConsensusFilter, req.ReplicationFactor)
 	if err != nil {
-		return result, err
-	}
-	filterResult, err := ApplyFilter(filter, reveals)
-	result.consensus = filterResult.Consensus
-	result.proxyPubKeys = filterResult.ProxyPubKeys
-	if err != nil {
-		return result, k.logErrAndRet(ctx, err, types.ErrApplyingFilter, req)
-	}
+		result.result = []byte(err.Error())
+		result.exitInfo.ExitCode = TallyExitCodeInvalidFilterInput
+	} else {
+		filterResult, err = ApplyFilter(filter, reveals)
+		result.consensus = filterResult.Consensus
+		result.proxyPubKeys = filterResult.ProxyPubKeys
 
-	// Phase II: Tally Program Execution
-	vmRes, err := k.ExecuteTallyProgram(ctx, req, filterResult, reveals)
-	if err != nil {
-		return TallyResult{}, err
+		// Phase II: Tally Program Execution
+		if err != nil {
+			result.result = []byte(err.Error())
+			result.exitInfo.ExitCode = TallyExitCodeFilterError
+		} else {
+			vmRes, err := k.ExecuteTallyProgram(ctx, req, filterResult, reveals)
+			if err != nil {
+				result.result = []byte(err.Error())
+				result.exitInfo.ExitCode = TallyExitCodeFilterError
```

Comment:
Same here. Not sure if there's a scenario where the tally consumes gas but still returns an error, but just to be sure we should probably set it similar to the success case.